sphinx/source/docs/dev_guide/creating_pull_requests.rst

```diff
@@ -0,0 +1,153 @@
+
+    For example:
+
+    ``11423_table_column_visible``
+
+    See `BEP 6: Branching Strategy`_ for more information on the different
+    branches in the Bokeh repository.
+
+3. Publish your branch and push a first commit
+    Before you can open a pull request, you first need to publish your branch.
+    At this time, you should also include a first commit.
+
+4. Open a new pull request
+    After publishing your branch and adding a first commit, go to the
+    `Bokeh repository`_ on GitHub. GitHub should have detected your recent
+    updates to your branch. In this case, GitHub will suggest opening a pull
+    request for you. If not, switch to the tab "Pull request" in the top menu
+    and use the button "New pull request".
+
+    To create a pull request, make sure to select the default branch you used
+    when creating your branch as ``base`` (eg. ``branch-2.4``). Select your
+    branch as ``compare``.
+
+5. Write your pull request
+    After creating your pull request, GitHub will compare your branch to the
+    base branch and highlight all your proposed changes.
+
+    First, enter a title for your pull request. This title should make clear
+    what your pull request does. For example: "Fix PNG export", "Add panel to
+    tests", or "Document SVG backend".
+
+    Next, enter a description. Include some background about what your pull
+    request does and why you decided to write things the way you did. Also, link
+    to the issue your pull request is based on. To do so, use a keyword such as
+    "fixes," followed by the number of the issue. For example "Fixes #11479".
+    See `Linking a pull request to an issue using a keyword`_ in the
+    `GitHub documentation`_ for more information. Your description should also
+    include information about :ref:`tests <devguide_writing_tests>` and
+    :ref:`documentation <devguide_documentation>`, if applicable.
+
+6. Add more commits to your pull request
+    Once you have created a pull request, keep adding commits. A member of the
+    `Bokeh core team`_ will review your pull request and help you along the way.
+    The team member will also update any `labels in your pull request <wiki pull requests_>`_.
+    Reviewing pull requests can be time-consuming, so be aware that it might
+    take a while to receive feedback.
+
+    Once you and a member of the `Bokeh core team`_ agree that your pull request
+    is ready to be merged, the team member will update the labels on your pull
+    request and merge your changes into the repository.
+
+7. Look for the next issue to work on
+    With your first pull request merged, you should take another look at Bokeh's
+    `issue tracker`_ to find the next issue to work on.
+
+    Once your successfully complete two substantive pull requests, you are
+    eligible to become a member of the `Bokeh development team`_. See
+    `BEP 4: Project Roles`_ for more information about all roles in the Bokeh
+    project.
```

Comment:
I'm not sure it's something to mention here but just for record one reason it is good for folks to join `@bokeh/dev` is that then they can work from a clone of the main repo instead of a fork. AFAIK this is still necessary for the examples test report to be able to be published. It also makes it much easier for other core people to check out the branch (and when necessary and appropriate collaborate directly by pushing to the branch)